src/thor/bidirectional_astar.cc

```diff
@@ -97,10 +111,33 @@ void BidirectionalAStar::Init(const PointLL& origll, const PointLL& destll) {
   hierarchy_limits_reverse_ = costing_->GetHierarchyLimits();
 }
 
+// Container for the data we iterate over in Expand* function
+struct EdgeMetadata {
+  const DirectedEdge* edge;
+  GraphId edge_id;
+  EdgeStatusInfo* edge_status;
+
+  static EdgeMetadata make(const GraphId& node,
+                           const NodeInfo* nodeinfo,
+                           const GraphTile* tile,
+                           EdgeStatus& edge_status_) {
+    GraphId edge_id = {node.tileid(), node.level(), nodeinfo->edge_index()};
+    EdgeStatusInfo* edge_status = edge_status_.GetPtr(edge_id, tile);
+    const DirectedEdge* directededge = tile->directededge(edge_id);
+    return {.edge = directededge, .edge_id = edge_id, .edge_status = edge_status};
```

Comment:
The designated initializer is C++20 features. You need to make it C++11 using list or direct initialization.

Reply:
this seems to be an interesting exception! apparently in c++11 "trivial" designated intializers are supported. "trivial" apparently means you specify all members of the struct and all are in the same order as they are in the structs declaration. so for example this works even without c++11 at least in gcc:

```
struct foo{
  int bar;
  int baz;
};
int main(){
  foo f={.bar=3,.baz=7};
  return 0;
}
```
but this does not:

```
struct foo{
  int bar;
  int baz;
};
int main(){
  foo f={.baz=3,.bar=7};
  return 0;
}
```

In the end i do agree its probably not a good idea to start using this syntax if we can only use special case. i'm pretty sure this would also be the only example of this usage in the entire project. i know its less clear but without the designators is probably the way to go

Reply:
@kevinkreiser 
> apparently in c++11 and earlier "trivial" designated intializers are supported. 

No, it is not, I'm afraid.

What you are observing with GCC and clang is C99 feature that kicks in by default.
If C99 is disabled e.g. `-std=c++11 -pedantic` then that program will fail to compile 

- GCC 9

```
warning: C++ designated initializers only available with '-std=c++2a' or '-std=gnu++2a' [-Wpedantic]
   foo f={.bar=3,.baz=7};
          ^
```

- clang 9

```
warning: designated initializers are a C99 feature [-Wc99-extensions]
  foo f={.bar=3,.baz=7};
         ^~~~~~
```

MSVC does not support C99 features for C itself. It only supports C99 features that are part of C++11 or later versions of C++.

Reply:
yeah that is why i said "at least in gcc" because i had suspected it wasnt the case with all compilers. thanks for the confirmation!

Reply:
Adjusted code.